views.py

```diff
@@ -425,3 +444,160 @@ def edit_supporter_band(request, supporter_id=None):
+    return render(request, template, context)
+
+
+@require_POST
+def recommend_us_search_journal(request):
+    context = {
+        'role': 'editor',
+        'step': '4',
+    }
+    template = hourglass('custom/recommend-us-search-journal.html')
+    return render(request, template, context)
+
+
+@require_POST
+def recommend_us_choose_article(request):
+    query = request.POST.get('article_query', '')
+    if query:
+        articles = request.press.published_articles.filter(title__icontains=query)
+    else:
+        articles = submission_models.Article.objects.none()
+    role = request.POST.get('role', '')
+    template = hourglass('custom/recommend-us-choose-article.html')
+    context = {
+        'articles': articles,
+        'step': '5',
+        'role': role,
+        'query': query,
+    }
+    return render(request, template, context)
+
+
+@require_POST
+def recommend_us_choose_journal(request):
+    query = request.POST.get('journal_query', '')
+    if query:
+        journal_names = SettingValue.objects.filter(
+            setting__name='journal_name',
+            value__icontains=query,
+            journal__in=request.press.public_journals,
+        )
+    else:
+        journal_names = SettingValue.objects.none()
+    role = request.POST.get('role', '')
+    template = hourglass('custom/recommend-us-choose-journal.html')
+    context = {
+        'journal_names': journal_names,
+        'step': '5',
+        'role': role,
+        'query': query,
+    }
+    return render(request, template, context)
+
+
+@require_POST
+def recommend_us_generate_email(request):
+    role = request.POST.get('role', '')
+    step = '6'
+    setting_name = 'recommend_us_general_email'
+
+    if role == 'author':
+        setting_name = 'recommend_us_author_email'
+    elif role == 'reader':
+        setting_name = 'recommend_us_reader_email'
+    elif role == 'editor':
+        setting_name = 'recommend_us_editor_email'
+    else:
+        # We assume this is "other" because no role input is available
+        # in the request data.
+        setting_name = 'recommend_us_general_email'
+        step = '4'
+
+    # An author, reader, or editor could not find the article or journal so
+    # selects general
+    if 'general' in request.POST:
+        setting_name = 'recommend_us_general_email'
+
+    template = Template(utils.setting(setting_name))
+    setting_context = RequestContext(request)
+
+    article_pk = request.POST.get('article_pk', '')
+    if article_pk:
+        setting_context.push(
+            {
+                'article': get_object_or_404(
+                    submission_models.Article,
+                    pk=article_pk,
+                )
+            }
+        )
+
+    journal_pk = request.POST.get('journal_pk', '')
+    if journal_pk:
+        setting_context.push(
+            {
+                'journal': get_object_or_404(
+                    Journal,
+                    pk=journal_pk,
+                )
```

Comment:
I think we also want the filter you used previously:
`journal__in=request.press.public_journals,`